utils/kubernetes/service.go

```diff
@@ -2,109 +2,74 @@ package kubernetes
 
 import (
 	"context"
+	"strings"
 
 	"github.com/layer5io/meshkit/utils"
-	coreV1 "k8s.io/api/core/v1"
-	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
 )
 
-// GetServiceEndpoint returns the endpoint for the given service
-func (client *Client) GetServiceEndpoint(ctx context.Context, svcName, namespace string) (*utils.Endpoint, error) {
-	svc, err := client.KubeClient.CoreV1().Services(namespace).Get(ctx, svcName, v1.GetOptions{})
-	if err != nil {
-		return nil, ErrServiceDiscovery(err)
-	}
-
-	// Try loadbalancer endpoint
-	if endpoint := extractLoadBalancerEndpoint(svc); endpoint != nil {
-		return endpoint, nil
-	}
+type ServiceOptions struct {
+	Name         string
+	Namespace    string
+	PortSelector string
+	PortNumber   *int32
+	APIServerURL string
```

Comment:
what's that? would be nice with a comment :smile:

Reply:
Ah yes okay adding them